### PR TITLE
1033 convert appd schema to json

### LINKED ISSUE #1033: App Directory schema contains errors
### Area of Issue
[x] App Directory

### Issue Description: 
The app directory schema is currently failing its test. The example schema at https://github.com/InteropIO/FDC3/blob/544-Desktop-Agent_bridging-Proposal-mk2/src/app-directory/specification/examples/application/fdc3-workbench.json contains error (type should be "web" rather than "browser". However, once corrected the test still won't pass as the `details` element can't be validated (due to the use of `oneOf`, where the subschemas overap - this should be `anyOf`), amongst other errors.

Errors are only returned one at a time due to the encoding in YAML, switching to JSON schema would allow all errors to output at once. 

Correct the errors in the App Directory schema and consider switching the encoding to JSON Schema.

### PR DESCRIPTION
**Pull request by kriswest**
resolves #1033 
builds on PR #1034 

Converts the AppD schema file from YAML to JSON Schema (where the tooling is a little better and ensuring that we use one schema format across FDC3, where there are already other examples of JSON Schema use for contexts, bridging and bridging asyncAPI).

Updates were applied to to package scripts and the website to use the updated encoding.

### CODE REVIEW

**Comment by WatsonCIQ** [comment]
Works for me 👍

File: src/app-directory/specification/appd.schema.json [hunk 1]
```diff
@@ -0,0 +1,1523 @@
...
+		  "enum": [
+			"web",
+			"native",
+			"citrix",
+			"onlineNative",
+			"other"
+		  ]
+		},
+		"LaunchDetails": {
+		  "description": "The type specific launch details of the application. These details are intended to be  vendor-agnostic and MAY be duplicated or overridden by details provided in the hostManifests  object for a specific host.",
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:650]:
```suggestion
		  "description": "The type specific launch details of the application. These details are intended to be vendor-agnostic and MAY be duplicated or overridden by details provided in the hostManifests object for a specific host.",
```

File: src/app-directory/specification/appd.schema.json [hunk 2]
```diff
@@ -0,0 +1,1523 @@
...
+			  "type": "string",
+			  "description": "Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary as it is defined in the app's AppD record.",
+			  "deprecated": true
+			},
+			"contexts": {
+			  "type": "array",
+			  "items": {
+				"type": "string"
+			  },
+			  "description": "A comma separated list of the types of contexts the intent offered by the application can process,   where the first part of the context type is the namespace e.g.\"fdc3.contact, org.symphony.contact\""
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:779]:
```suggestion
			  "description": "A comma separated list of the types of contexts the intent offered by the application can process, where the first part of the context type is the namespace e.g.\"fdc3.contact, org.symphony.contact\""
```

File: src/app-directory/specification/appd.schema.json [hunk 3]
```diff
@@ -0,0 +1,1523 @@
...
+			},
+			"resultType": {
+			  "type": "string",
+			  "description": "An optional type for output returned by the application, if any, when resolving this intent.  May indicate a context type by type name (e.g. \"fdc3.instrument\"), a channel (e.g. \"channel\")  or a combination that indicates a channel that returns a particular context type  (e.g. \"channel<fdc3.instrument>\")."
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:783]:
```suggestion
			  "description": "An optional type for output returned by the application, if any, when resolving this intent. May indicate a context type by type name (e.g. \"fdc3.instrument\"), a channel (e.g. \"channel\") or a combination that indicates a channel that returns a particular context type (e.g. \"channel<fdc3.instrument>\")."
```

File: src/app-directory/specification/appd.schema.json [hunk 4]
```diff
@@ -0,0 +1,1523 @@
...
+			  "description": "Describes the application's use of App Channels.\n\nThis metadata is not currently used by the desktop agent, but is provided to help find apps \nthat will interoperate with this app and to document API interactions for use by other app \ndevelopers.\n",
+			  "items": {
+				"type": "object",
+				"required": [
+				  "id"
+				],
+				"properties": {
+				  "id": {
+					"type": "string",
+					"description": "The id of the App Channel.  N.b. in FDC3 2.0 this field was incorrectly  called `name`.\n"
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:853]:
```suggestion
					"description": "The id of the App Channel. N.b. in FDC3 2.0 this field was incorrectly called `name`.\n"
```

File: src/app-directory/specification/appd.schema.json [hunk 5]
```diff
@@ -0,0 +1,1523 @@
...
+			"displayName": {
+			  "type": "string",
+			  "description": "An optional display name for the intent that may be used in UI instead of the name."
+			},
+			"contexts": {
+			  "type": "array",
+			  "items": {
+				"type": "string"
+			  },
+			  "description": "A comma separated list of the types of contexts the intent offered by the application can process.  where the first part of the context type is the namespace e.g.\"fdc3.contact, org.symphony.contact\""
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:917]:
```suggestion
			  "description": "A comma separated list of the types of contexts the intent offered by the application can process, where the first part of the context type is the namespace e.g.\"fdc3.contact, org.symphony.contact\""
```

File: src/app-directory/specification/appd.schema.json [hunk 6]
```diff
@@ -0,0 +1,1523 @@
...
+			},
+			"customConfig": {
+			  "deprecated": true,
+			  "type": "object",
+			  "description": "Custom configuration for the intent that may be required for a particular desktop agent. Deprecated due to a lack of defined use cases."
+			}
+		  }
+		},
+		"ApplicationV1": {
+		  "description": "(Deprecated v1 API version) Defines an application retrieved from an FDC3 App Directory, which can then be launched. Launching typically means running for a user on a desktop.  The details around 'launching' including who or what might do it, and how the launch action is initiated are discussed elsewhere in the FDC3 App Directory spec.\n",
```
Comment (hughtroeger) [inline src/app-directory/specification/appd.schema.json:927]:
```suggestion
		  "description": "(Deprecated v1 API version) Defines an application retrieved from an FDC3 App Directory, which can then be launched. Launching typically means running for a user on a desktop. The details around 'launching' including who or what might do it, and how the launch action is initiated are discussed elsewhere in the FDC3 App Directory spec.\n",
```

**Review by hughtroeger — APPROVED** [review]
Comments are about double spaces. Was a little unsure about all the newlines in the description fields. Some clearly delineate mini paragraphs, but some occur mid sentence and some have trailing newlines while others do not. Anyway all whitespace issues, content looks good.

**Comment by kriswest** [comment]
>Comments are about double spaces. Was a little unsure about all the newlines in the description fields. Some clearly delineate mini paragraphs, but some occur mid sentence and some have trailing newlines while others do not. Anyway all whitespace issues, content looks good.

I suspect all of these were existing in the YAML file, which is an awkward format unless you're really used to it. A good reason to convert to JSON Schema! Edit: checked and yes double spacing and line breaks were in YAML. 

Interestingly none were rendered. This is because all the copy is encoded in markdown, so single line breaks and double spaces are meaningless and drop out.  However, double linebreaks or those connected to other markdown syntax (e.g. bulleted lists) did have meaning. All should now be correct.

Preview: https://deploy-preview-1035--fdc3.netlify.app/schemas/next/app-directory.html

**Review by mattjamieson — APPROVED** [review]

**Pull request merged**